### PR TITLE
Blog app: add authorization rules

### PR DESCRIPTION
**Pull request by olawale-o**
## In this PR, we did the following:

- Install `CanCanCan` in your project.
- Add a `role` column to the users table. Remember to use a migration for this.
- A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
   - For that you need to implement the post deleting functionality. Add the "Delete" button **(we used a trash icon)** to the view and make sure that only authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin role (column `role` has value `"admin"`). Use CanCanCan for this authorization.
  - For that you need to implement the comment deleting functionality. Add the "Delete" button **(we used a trash icon)** to the view and make sure that only authorized users can see it.

## Please Note:
- We used a **letter_opener** gem to preview the confirmation mail in the browser.
- You have to create the user with `admin` privileges via the terminal or command prompt with the following attributes:
    - name, email, password, role
  - `role` must be set to **admin**
  - bio **(optional)**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -3,6 +3,8 @@ git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
 ruby '3.0.0'
 
+# use cancancan
```
Comment (arslanbisharat) [inline Gemfile:6]: - [ ] Make sure you have rswag gem installed. 
```
gem 'rswag-api'
gem 'rswag-ui'

```
```
group :development, :test do ... 
gem 'rswag-specs'
 end 
```

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: Gemfile (hunk 1 above)
Comment (olawale-o) [reply Gemfile:6]: Hi @arslanbisharat 
Please I would like to know why you asked us to install this gems. I believe they are not part of the milestone requirements

Comment (olawale-o) [reply Gemfile:6]: Hi @arslanbisharat
Although we have installed them

Comment (Emmyn5600) [reply Gemfile:6]: Hello, you can remove those gems, we don't need it at this moment.

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi, @olawale-o and @PolinaStamenova 

Good job so far! 👏🏼 
- The authorization has been implemented
- Send a confirmation message to the email after a user signs up.
- Create, delete has been implemented.
- Add and delete comment has been implemented.

Great work on making the project requirements :clap:
You've done well implementing the project requirements, but there are still some issues which aren't been addressed yet 
 
## STATUS: CHANGES REQUESTED :recycle:


### Changes requested.

- [ ] we are missing  `edit post` implementation
- Optional suggestion, instead of displaying the user as a text, let us try to add a user's profile for someone who signed in.
cancan-2

 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Comment by olawale-o** [comment]
Hi @Emmyn5600 

- Post edit is not part of the requirement
- Please check the requirements and the previous one. Post editing was never part of the requirements

**Comment by Emmyn5600** [comment]
Hi @olawale-o, I see, let me change it as optional, so you can submit it for the re-review.

**Comment by olawale-o** [comment]
Hi @Emmyn5600 
Could you please!!! check the requirements again and I would be happy if you could point me to where post edit was part of the requirement

**Comment by olawale-o** [comment]
Hi @Emmyn5600 
The user you see next to the logout button is used to indicate whether it is an admin or a regular user

**Comment by olawale-o** [comment]
> Hi @olawale-o, I see, let me change it as optional, so you can submit it for the re-review.

Ok. Thank you for this. I will resubmit

File: Gemfile (hunk 1 above)
Comment (olawale-o) [reply Gemfile:6]: I will remove the gems again

**Review by Emmyn5600 — APPROVED** [review]
Hi Team 

Good job👏🏼

## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:
Your project is complete!  It's time to merge it :shipit:
Great work on implementing the changes requested by the previous reviewers.

 #### Optional suggestions
 -  N/A.
 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Comment by olawale-o** [comment]
Hi @Emmyn5600 
Thank you for your time and effort on the review. We really appreciate your quick response

**Pull request merged**